Alignment control: Dropdown throws an exception upon opening

<!--
Please fill out ALL required sections. Bug reports with missing information will
be closed.

Before submitting a bug report:

- Check if the bug has already been fixed by updating WordPress and/or Gutenberg.
- Check if the bug is caused by a plugin by deactivating all plugins except Gutenberg.
- Check if the bug is caused by a theme by activating a default theme e.g. Twenty Twenty.
- Check if the bug has already been reported by searching https://github.com/WordPress/gutenberg/issues.

If this is a security issue, please report it in HackerOne instead:
https://hackerone.com/wordpress
-->

## Description
<!-- Please write a brief description of the bug. -->

An exception is thrown when opening the block alignment controls on most blocks, the only one that is currently working is the Image block.

**Exception:**
```
ExceptionsManager.js:180 Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.

Check the render method of `Dropdown`.

This error is located at:
    in RCTView (at View.js:34)
    in View (at TouchableHighlight.js:289)
    in TouchableHighlight (at TouchableHighlight.js:352)
    in Unknown (at bottom-sheet/index.native.js:578)
    in RCTScrollContentView (at ScrollView.js:1107)
    in RCTScrollView (at ScrollView.js:1238)
    in ScrollView (at ScrollView.js:1264)
    in ScrollView (at bottom-sheet/index.native.js:554)
    in RCTView (at View.js:34)
    in View (at keyboard-avoiding-view.native.js:118)
    in KeyboardAvoidingView (at bottom-sheet/index.native.js:528)
    in RCTView (at View.js:34)
    in View (at createAnimatedComponent.js:217)
    in AnimatedComponent (at createAnimatedComponent.js:278)
    in AnimatedComponentWrapper (at createAnimatableComponent.js:599)
    in withAnimatable(View) (created by ReactNativeModal)
    in RCTView (at View.js:34)
    in View (at AppContainer.js:107)
    in RCTView (at View.js:34)
    in View (at AppContainer.js:134)
    in AppContainer (at Modal.js:228)
    in RCTView (at View.js:34)
    in View (at Modal.js:249)
    in VirtualizedListContextResetter (at Modal.js:247)
    in RCTModalHostView (at Modal.js:234)
    in Modal (created by ReactNativeModal)
    in ReactNativeModal (at bottom-sheet/index.native.js:499)
    in BottomSheet (at with-preferred-color-scheme/index.native.js:30)
    in WithPreferredColorScheme(BottomSheet) (at dropdown-menu/index.native.js:116)
    in Dropdown (at dropdown-menu/index.native.js:68)
    in DropdownMenu (at with-preferred-color-scheme/index.native.js:30)
    in WithPreferredColorScheme(DropdownMenu) (at toolbar-dropdown-menu/index.js:17)
    in ToolbarDropdownMenu (at ui.js:158)
    in BlockAlignmentUI (at block-alignment-control/index.js:7)
    in BlockAlignmentControl (at align.js:147)
    in RCTView (at View.js:34)
    in View (at toolbar-group-container.native.js:21)
    in ToolbarGroupContainer (at with-preferred-color-scheme/index.native.js:30)
    in WithPreferredColorScheme(ToolbarGroupContainer) (at toolbar-group/index.js:94)
    in ToolbarGroup (at slot.native.js:29)
    in SlotComponent (at slot.js:92)
    in Slot (at slot-fill/index.native.js:16)
    in Slot (at slot-fill/index.native.js:23)
    in BlockControlsBlockSlot (at slot.native.js:24)
    in BlockControlsSlot (at block-toolbar/index.native.js:44)
    in BlockToolbar (at header-toolbar/index.native.js:100)
    in RCTScrollContentView (at ScrollView.js:1107)
    in RCTScrollView (at ScrollView.js:1238)
    in ScrollView (at ScrollView.js:1264)
    in ScrollView (at header-toolbar/index.native.js:89)
    in RCTView (at View.js:34)
    in View (at header-toolbar/index.native.js:83)
    in HeaderToolbar (at with-preferred-color-scheme/index.native.js:30)
    in WithPreferredColorScheme(HeaderToolbar) (at with-select/index.js:57)
    in Unknown (at pure/index.tsx:42)
    in WithViewportMatch(WithPreferredColorScheme(HeaderToolbar)) (at with-dispatch/index.js:96)
    in WithDispatch(WithViewportMatch(WithPreferredColorScheme(HeaderToolbar))) (at with-select/index.js:57)
    in Unknown (at pure/index.tsx:42)
    in WithSelect(WithDispatch(WithViewportMatch(WithPreferredColorScheme(HeaderToolbar)))) (at header/index.native.js:49)
    in Header (at layout/index.native.js:171)
    in RCTView (at View.js:34)
    in View (at keyboard-avoiding-view/index.ios.js:123)
    in RCTView (at View.js:34)
    in View (at KeyboardAvoidingView.js:209)
    in KeyboardAvoidingView (at createAnimatedComponent.js:217)
    in AnimatedComponent (at createAnimatedComponent.js:278)
    in AnimatedComponentWrapper (at keyboard-avoiding-view/index.ios.js:104)
    in KeyboardAvoidingView (at layout/index.native.js:160)
    in RCTSafeAreaView (at SafeAreaView.js:51)
    in SafeAreaView (at layout/index.native.js:137)
    in RCTView (at View.js:34)
    in View (at tooltip/index.native.js:268)
    in TooltipSlot (at layout/index.native.js:136)
    in Layout (at with-preferred-color-scheme/index.native.js:30)
    in WithPreferredColorScheme(Layout) (at with-select/index.js:57)
    in Unknown (at pure/index.tsx:42)
    in WithSelect(WithPreferredColorScheme(Layout)) (at editor.native.js:173)
    in BlockRefsProvider (at provider/index.native.js:28)
    in BlockEditorProvider (at with-registry-provider.js:24)
    in Unknown (at with-registry/index.js:23)
    in WithRegistryProvider(BlockEditorProvider) (at provider/index.js:108)
    in BlockContextProvider (at provider/index.js:107)
    in EntityProvider (at provider/index.js:106)
    in EntityProvider (at provider/index.js:105)
    in EditorProvider (at with-registry-provider.js:27)
    in Unknown (at with-registry/index.js:23)
    in WithRegistryProvider(EditorProvider) (at provider/index.native.js:337)
    in NativeEditorProvider (at with-dispatch/index.js:96)
    in WithDispatch(NativeEditorProvider) (at with-select/index.js:57)
    in Unknown (at pure/index.tsx:42)
    in WithSelect(WithDispatch(NativeEditorProvider)) (at editor.native.js:166)
    in SlotFillProvider (at editor.native.js:165)
    in Editor (at with-dispatch/index.js:96)
    in WithDispatch(Editor) (at with-select/index.js:57)
    in Unknown (at pure/index.tsx:42)
    in WithSelect(WithDispatch(Editor)) (at src/index.native.js:31)
    in App (at renderApplication.js:47)
    in RCTView (at View.js:34)
    in View (at AppContainer.js:107)
    in RCTView (at View.js:34)
    in View (at AppContainer.js:134)
    in AppContainer (at renderApplication.js:40)
```

## Step-by-step reproduction instructions
<!--
Please list the steps needed to reproduce the bug. For example:
1. Go to '...'
2. Click on '...'
3. Scroll down to '...'
-->
1. Add a Cover block.
2. Tap on the alignment control button located in the toolbar.
<img width="45" alt="Screenshot 2021-09-28 at 10 36 07" src="https://user-images.githubusercontent.com/14905380/135106609-07f4b286-7474-44ef-bc3a-11bf97570bca.png">
3. Observe that an exception is thrown.

## Expected behaviour
<!-- Please describe what you expected to happen. -->
The alignment control shouldn't throw any exception.

## Actual behaviour
<!-- Please describe what actually happened. -->
The alignment control is not working and throws an exception.

## Screenshots or screen recording (optional)
<!--
If possible, please upload a screenshot or screen recording which demonstrates
the bug. 
-->


https://user-images.githubusercontent.com/14905380/135053336-f39b6d92-3e2b-4557-8873-748e1b302b81.mp4



## WordPress information
- WordPress version: <!-- e.g. "5.8.0". Find this in Tools → Site Health → Info → WordPress -->N/A
- Gutenberg version: <!-- e.g. "9.4.0" or "Not installed" -->N/A
- Are all plugins except Gutenberg deactivated? <!-- "Yes" or "No" -->N/A
- Are you using a default theme (e.g. Twenty Twenty-One)? <!-- "Yes" or "No" -->N/A

## Device information
- Device: <!-- e.g. "Pixel 4" or "iPhone 11" -->iPhone 12 Pro Max
- Operating system: <!-- e.g. "Android 11.0" or "iOS 14.0" -->iOS 14.5
- WordPress app version: <!-- e.g. "16.3" or branch name / git commit hash -->develop

--- a/packages/block-editor/src/components/block-alignment-control/test/index.native.js
+++ b/packages/block-editor/src/components/block-alignment-control/test/index.native.js
@@ -1,0 +1,37 @@
+/**
+ * External dependencies
+ */
+import { render, fireEvent } from 'test/helpers';
+
+/**
+ * Internal dependencies
+ */
+import BlockAlignmentUI from '../ui';
+
+it( 'should call onChange with undefined, when the control is already active', () => {
+	const onChangeMock = jest.fn();
+	const screen = render(
+		<BlockAlignmentUI value="left" onChange={ onChangeMock } />
+	);
+	const alignButton = screen.getByA11yLabel( 'Align' );
+	fireEvent.press( alignButton );
+	const noneAlignmentButton = screen.getByA11yLabel( 'None' );
+	fireEvent.press( noneAlignmentButton );
+
+	expect( onChangeMock ).toHaveBeenCalledTimes( 1 );
+	expect( onChangeMock ).toHaveBeenCalledWith( undefined );
+} );
+
+it( 'should call onChange with alignment value when the control is inactive', () => {
+	const onChangeMock = jest.fn();
+	const screen = render(
+		<BlockAlignmentUI value="left" onChange={ onChangeMock } />
+	);
+	const alignButton = screen.getByA11yLabel( 'Align' );
+	fireEvent.press( alignButton );
+	const noneAlignmentButton = screen.getByA11yLabel( 'Align center' );
+	fireEvent.press( noneAlignmentButton );
+
+	expect( onChangeMock ).toHaveBeenCalledTimes( 1 );
+	expect( onChangeMock ).toHaveBeenCalledWith( 'center' );
+} );

--- a/packages/block-editor/src/components/block-alignment-control/test/index.native.js
+++ b/packages/block-editor/src/components/block-alignment-control/test/index.native.js
@@ -8,7 +8,7 @@ import { render, fireEvent } from 'test/helpers';
  */
 import BlockAlignmentUI from '../ui';
 
-it( 'should call onChange with undefined, when the control is already active', () => {
+it( 'should call onChange with undefined when the control is already active', () => {
 	const onChangeMock = jest.fn();
 	const screen = render(
 		<BlockAlignmentUI value="right" onChange={ onChangeMock } />

--- a/packages/block-editor/src/components/block-alignment-control/test/index.native.js
+++ b/packages/block-editor/src/components/block-alignment-control/test/index.native.js
@@ -11,12 +11,12 @@ import BlockAlignmentUI from '../ui';
 it( 'should call onChange with undefined, when the control is already active', () => {
 	const onChangeMock = jest.fn();
 	const screen = render(
-		<BlockAlignmentUI value="left" onChange={ onChangeMock } />
+		<BlockAlignmentUI value="right" onChange={ onChangeMock } />
 	);
 	const alignButton = screen.getByA11yLabel( 'Align' );
 	fireEvent.press( alignButton );
-	const noneAlignmentButton = screen.getByA11yLabel( 'None' );
-	fireEvent.press( noneAlignmentButton );
+	const rightAlignmentButton = screen.getByA11yLabel( 'Align right' );
+	fireEvent.press( rightAlignmentButton );
 
 	expect( onChangeMock ).toHaveBeenCalledTimes( 1 );
 	expect( onChangeMock ).toHaveBeenCalledWith( undefined );
@@ -29,8 +29,8 @@ it( 'should call onChange with alignment value when the control is inactive', ()
 	);
 	const alignButton = screen.getByA11yLabel( 'Align' );
 	fireEvent.press( alignButton );
-	const noneAlignmentButton = screen.getByA11yLabel( 'Align center' );
-	fireEvent.press( noneAlignmentButton );
+	const centerAlignmentButton = screen.getByA11yLabel( 'Align center' );
+	fireEvent.press( centerAlignmentButton );
 
 	expect( onChangeMock ).toHaveBeenCalledTimes( 1 );
 	expect( onChangeMock ).toHaveBeenCalledWith( 'center' );

--- a/packages/block-editor/src/components/block-alignment-control/ui.js
+++ b/packages/block-editor/src/components/block-alignment-control/ui.js
@@ -21,6 +21,7 @@ import {
 	stretchFullWidth,
 	stretchWide,
 } from '@wordpress/icons';
+import { Platform } from '@wordpress/element';
 
 /**
  * Internal dependencies
@@ -91,69 +92,72 @@ function BlockAlignmentUI( {
 		label: __( 'Align' ),
 		toggleProps: { describedBy: __( 'Change alignment' ) },
 	};
-	const extraProps = isToolbar
-		? {
-				isCollapsed,
-				controls: enabledControls.map( ( { name: controlName } ) => {
-					return {
-						...BLOCK_ALIGNMENTS_CONTROLS[ controlName ],
-						isActive:
-							value === controlName ||
-							( ! value && controlName === 'none' ),
-						role: isCollapsed ? 'menuitemradio' : undefined,
-						onClick: () => onChangeAlignment( controlName ),
-					};
-				} ),
-		  }
-		: {
-				children: ( { onClose } ) => {
-					return (
-						<>
-							<MenuGroup className="block-editor-block-alignment-control__menu-group">
-								{ enabledControls.map(
-									( { name: controlName, info } ) => {
-										const {
-											icon,
-											title,
-										} = BLOCK_ALIGNMENTS_CONTROLS[
-											controlName
-										];
-										// If no value is provided, mark as selected the `none` option.
-										const isSelected =
-											controlName === value ||
-											( ! value &&
-												controlName === 'none' );
-										return (
-											<MenuItem
-												key={ controlName }
-												icon={ icon }
-												iconPosition="left"
-												className={ classNames(
-													'components-dropdown-menu__menu-item',
-													{
-														'is-active': isSelected,
-													}
-												) }
-												isSelected={ isSelected }
-												onClick={ () => {
-													onChangeAlignment(
-														controlName
-													);
-													onClose();
-												} }
-												role="menuitemradio"
-												info={ info }
-											>
-												{ title }
-											</MenuItem>
-										);
-									}
-								) }
-							</MenuGroup>
-						</>
-					);
-				},
-		  };
+	const extraProps =
+		isToolbar || Platform.isNative
+			? {
+					isCollapsed: isToolbar ? isCollapsed : undefined,
+					controls: enabledControls.map(
+						( { name: controlName } ) => {
+							return {
+								...BLOCK_ALIGNMENTS_CONTROLS[ controlName ],
+								isActive:
+									value === controlName ||
+									( ! value && controlName === 'none' ),
+								role: isCollapsed ? 'menuitemradio' : undefined,
+								onClick: () => onChangeAlignment( controlName ),
+							};
+						}
+					),
+			  }
+			: {
+					children: ( { onClose } ) => {
+						return (
+							<>
+								<MenuGroup className="block-editor-block-alignment-control__menu-group">
+									{ enabledControls.map(
+										( { name: controlName, info } ) => {
+											const {
+												icon,
+												title,
+											} = BLOCK_ALIGNMENTS_CONTROLS[
+												controlName
+											];
+											// If no value is provided, mark as selected the `none` option.
+											const isSelected =
+												controlName === value ||
+												( ! value &&
+													controlName === 'none' );
+											return (
+												<MenuItem
+													key={ controlName }
+													icon={ icon }
+													iconPosition="left"
+													className={ classNames(
+														'components-dropdown-menu__menu-item',
+														{
+															'is-active': isSelected,
+														}
+													) }
+													isSelected={ isSelected }
+													onClick={ () => {
+														onChangeAlignment(
+															controlName
+														);
+														onClose();
+													} }
+													role="menuitemradio"
+													info={ info }
+												>
+													{ title }
+												</MenuItem>
+											);
+										}
+									) }
+								</MenuGroup>
+							</>
+						);
+					},
+			  };
 
 	return <UIComponent { ...commonProps } { ...extraProps } />;
 }

--- a/test/native/__mocks__/styleMock.js
+++ b/test/native/__mocks__/styleMock.js
@@ -135,4 +135,7 @@ module.exports = {
 	buttonNoBg: {
 		color: 'orange',
 	},
+	isSelected: {
+		color: 'blue',
+	},
 };

